### PR TITLE
Interactive mode updates

### PR DESCRIPTION
**Pull request by gtjoseph**
These updates were designed to assist those using interactive
mode to tune antennas and SDR gain.

* Add options to display distance and bearing in interactive mode

  Distance and bearing instead of latitude and longitude can now be
  displayed in interactive mode using the following options to
  dump1090-fa and view1090-fa.

  --interactive-show-distance   Show aircraft distance and bearing
                                instead of aircraft lat/lon
                                (requires --lat and --lon)
  --interactive-distance-units  Distance units ('km', 'sm', 'nm')
                                (default: 'nm')"

  You have to specify a reference --lat and --lon for this to
  work of course.

* A new line now shows at the top of the interactive display that
  has for the current sample:

  Total valid aircraft count
  Vidible aircraft count
     Will be less than total if the screen hasn't enough lines to show
     them all.
  Max RSSI
  Min RSSI
  Mean RSSI
  Max Distance

 Tot:  47 Vis:  47 RSSI: Max 25.4+ Mean -29.5 Min -36.9-  MaxD:  197.3nm+

* Add max distance and min/max RSSI indicators

  A '+' after the distance in a row indicates it's the row with the
  maximum distance.

  A '+' after the RSSI in a row indicates it's the row with the highest
  RSSI.

  A '-' after the RSSI in a row indicates it's the row with the lowest
  RSSI.

  The summary line at the top of the screen always shows the values for
  ALL aircraft, even those not visible.  The row indicators only mark
  visible rows though.

  In this example, the first aircraft is both the farthest away and has
  the weakest RSSI.  The second aircraft has the strongest RSSI.

```
 Tot:  47 Vis:  47 RSSI: Max 25.4+ Mean -29.5 Min -36.9-  MaxD:  197.3nm+     -
 Hex    Mode  Sqwk  Flight   Alt    Spd  Hdg  Dist(nm) Bearing  RSSI  Msgs  Ti
────────────────────────────────────────────────────────────────────────────────
 A8D5A4 S2                   34000  438  252   197.3+      85  -36.1-   26  2
 A39A13 S2ac  5740  FFT525   30750  439  256    98.8       68  -25.4+  123  0
 A70B23 S2ac  2744  LXJ553   43000  419  258   136.1       39  -33.6   174  0
```

* Finally, a new option '--interactive-callsign-filter' has been added
  to allow filtering interactive by callsign.  The value can be a
  simple string, in which case aircraft with that string anywhere in its
  callsign will be displayed, or a regular expression should you want a
  more precise match.

  Examples:
    --interactive-callsign-filter UAL
      will match all aircraft with UAL anywhere in its callsign.
    --interactive-callsign-filter "^UAL"
      will match only those callsigns that start with UAL.
    --interactive-callsign-filter "^(UAL|AAL)"
      will match only those callsigns that start with UAL or AAL.
    --interactive-callsign-filter "^N[0-9]" or "^N[[:digit:]]"
      will match only those callsigns that start with "N" and a number

  If you need more info on regular expressions, see this link:
  https://en.wikipedia.org/wiki/Regular_expression

### CODE REVIEW

File: interactive.c
Comment (mutability) [inline interactive.c:108]: REG_ICASE probably makes sense here too

File: interactive.c
Comment (mutability) [inline interactive.c:180]: Most of the time, a->callsign does not change. Can you cache the result of regexec here so it's not run on every refresh? (You'll need a "callsign changed" bit somewhere in aircraft state too)

File: interactive.c
Comment (mutability) [inline interactive.c:253]: Let's use `sizeof(strWhatever)` not a magic constant here (I know, the existing code is not good about doing that)

**Review by mutability — CHANGES_REQUESTED** [review]
Looks pretty good to me, some minor changes that would be nice to have.

A couple of general comments:

* is regex.h sufficiently universal? (From a quick look it's present on OS X , and has a BSD-derived manpage so it's probably available on *BSD too, so.. probably yes?)
* we're midway through a package release so it'll be a week or so before I can merge this

File: interactive.c
Comment (gtjoseph) [reply interactive.c:108]: Yep.  I thought about that, then forgot. :)

File: interactive.c
Comment (gtjoseph) [reply interactive.c:180]: Good idea.

File: interactive.c
Comment (gtjoseph) [reply interactive.c:253]: Yeah, it's hard to know what the policy is for a project you're new to so I  just default to copying the existing patterns.  :)
I'll fix.

**Comment by gtjoseph** [comment]
We use regex.h in our projects that also have to compile on bsd and macos.
https://asterisk.org
https://github.com/asterisk/asterisk

**Comment by gtjoseph** [comment]
Oops...  Don't merge this yet.  I think there's a display bug.

File: track.c
Comment (mutability) [inline track.c:1157]: I don't think you want this else branch. Most received messages do not contain a callsign so mm->callsign_valid is false, but that doesn't mean that the stored callsign has been invalidated.

**Review by mutability — CHANGES_REQUESTED** [review]

File: track.c
Comment (gtjoseph) [reply track.c:1157]: Hmmm.  Most messages from the same aircraft?  I didn't know that.    I'll remove it.

Comment (mutability) [reply track.c]: ADS-B transponders transmit "aircraft identification & category" messages (which carry the callsign) only once per 5 seconds (or once per 10 seconds when stationary on the ground). It's also carried in some Comm-B responses but those depend on SSR interrogation rates.

Meanwhile you might be getting 20+ messages/second of other types from the same aircraft.

File: interactive.c
Comment (mutability) [inline interactive.c:264]: This is going to be the geometric mean (since you're working in dB); not sure if that's what you intended?

**Review by mutability — APPROVED** [review]
Looks good to me, thanks! It will be a little while before I can merge this, need to wait for the 4.0 release to happen first.

File: interactive.c
Comment (gtjoseph) [reply interactive.c:264]: Yeah, I wanted it to be the simple mean of the displayed values otherwise it would take a whole bunch of explaining to folks. :)

**Comment by mutability** [comment]
Merged, thanks for these changes!

**Pull request merged**
